### PR TITLE
Bugfix: Age being overridden on old screens

### PR DESCRIPTION
**Pull request by kbrey**
What (if any) features are you implementing?

Fixes an issue where when a user returns to a screen from before age was replaced with birth year/month, the age displays as NaN, and when the user edits a non-household member question there's an error in the results. 
- Add back `age` to form data with deprecated comment for clarity
- Checks for undefined birth month/year and displays "N/A" // TODO: find a better translated default
- Update calcAge to default to `age` if `birthMonth` or `birthYear` are not defined

What (if anything) did you refactor?
- Pulled calcAge into `age.tsx` in Assets, as it is used outside of the HouseholdDataBlock
- Created `getCurrentMonthYear` helper function, used in `calcAge` and `AgeInput`
- Created `useFormatBirthMonthYear` hook to share code that translates and formats the birth month and year into `{month}/{year}`

### CODE REVIEW

File: src/Assets/age.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+export function useFormatBirthMonthYear() {
+  const translateNumber = useTranslateNumber()
+  return (birthMonth: number | null, birthYear: number | null) => {
+    if(birthMonth !== undefined && birthYear !== undefined) {
+      const formattedMonth = translateNumber(String(birthMonth).padStart(2, '0'));
+      const formattedYear = translateNumber(birthYear as number);
+      return `${formattedMonth}/${formattedYear}`;
+    }
+    return "N/A"
```
Comment (kbrey) [inline src/Assets/age.tsx:41]: @CalebPena is there a translated default that I can use?

Comment (CalebPena) [reply src/Assets/age.tsx:41]: If it does not exist, let's just not show it on the confirmation page.

**Review by CalebPena — APPROVED** [review]
Looks good. Can you pull from the dev branch. The confirmation cards got moved into a new file, so I think that you need to update that in the new file. The new file is called `/src/Components/Confirmation/ConfirmationSteps.tsx`. Let me know if you have any questions.

File: src/Assets/age.tsx [hunk 2]
```diff
@@ -11,18 +12,24 @@ export function getCurrentMonthYear() {
...
+export function hasBirthMonthYear(member: MemberBirthMonthYear | Partial<HouseholdData>): member is MemberBirthMonthYear {
+  const {birthMonth, birthYear} = member;
   const hasBirthYear = birthYear !== undefined && birthYear !== null;
   const hasBirthMonth = birthMonth !== undefined && birthMonth !== null;
   return hasBirthYear && hasBirthMonth;
 }
 
-export function calcAge(age: number, birthYear: number, birthMonth: number) {
-  const hasBirthDate = hasBirthMonthYear(birthYear, birthMonth);
+export function calcAge(member: Partial<HouseholdData>) {
```
Comment (CalebPena) [inline src/Assets/age.tsx:24]: Can you just use `HouseholdData` here?

**Review by CalebPena — APPROVED** [review]
Looks good. I added one optional suggestion.

File: src/Assets/age.tsx (hunk 2 above)
Comment (kbrey) [reply src/Assets/age.tsx]: @CalebPena Thanks for the suggestion, I cleaned up the types.

**Review by CalebPena — APPROVED** [review]

**Pull request merged**